File: Daily Challenge/January 2025/Count Prefix and Suffix Pairs I.cs
public class Solution {
  private bool isSuffixAndPrefix(string a, string b) {
    if (a.Length > b.Length) return false;

    for (int i = 0; i < a.Length; ++i) {
      if (a[i] != b[i] || a[i] != b[b.Length - a.Length + i]) return false;
    }

    return true;
  }

  public int CountPrefixSuffixPairs(string[] words) {
    int count = 0;

    for (int i = 0; i < words.Length; ++i) {
      for (int j = i + 1; j < words.Length; ++j) {
        if (isSuffixAndPrefix(words[i], words[j])) count++;
      }
    }

    return count;
  }
}

File: Daily Challenge/January 2025/Count Prefix and Suffix Pairs I.cpp
class Solution {
  bool isPrefixAndSuffix(const string& a, const string& b) {
    if (a.size() > b.size()) return false;

    for (int i = 0; i < a.size(); i++) {
      if (a[i] != b[i] or a[i] != b[b.size() - a.size() + i]) {
        return false;
      }
    }

    return true;
  }
public:
  int countPrefixSuffixPairs(vector<string>& words) {
    int count = 0;

    for (int i = 0; i < words.size(); ++i) {
      for (int j = i + 1; j < words.size(); ++j) {
        if (isPrefixAndSuffix(words[i], words[j])) count++;
      }
    }

    return count;
  }
};